src/arrayMethodReduce.js

```diff
@@ -5,7 +5,13 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = arguments.length > 1 ? initialValue : this[0];
+
+    for (let i = accumulator === this[0] ? 1 : 0; i < this.length; i++) {
```

Comment:
Same here. What if `this[0]` will accidentally be the same as `initialValue`? You will skip one iteration in that case